### PR TITLE
Milestone 2 

### PR DESCRIPTION
**Pull request by Guy-Gustave**
#### This milestone-2  pull request has the aim of :

- [X] Initializing Redux actions, reducers, and store 
- [X] Defining React components.

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Book = (props) => {
+  const { title, category } = props;
+  return (
```
Comment (teekaytech) [inline src/components/Book.js]: - [x] The Book.js component should accept <code>Book</code> object as prop <a href="https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/Book/AccceptBookObjectAsProp.png">screenshot from Pathwright</a>.

- [x] It should represent a single row of books with id, title, and category <a href="https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/Book/RepresentARowInBooksTable.png">screenshot from Pathwright</a>

- You do not have to break down the book properties at the prop level. You should pass the `book` object, which will be a representation of the book attributes (id, title, category)

File: src/containers/BooksList.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        <th>ID</th>
+        <th>Title</th>
+        <th>Category</th>
+      </tr>
+      {listBooks.map((book) => (
+        <tr key={book}>
+          <td>{book.id}</td>
+          <td>{book.title}</td>
+          <td>{book.category}</td>
+        </tr>
```
Comment (teekaytech) [inline src/containers/BooksList.js]: - [x] The `BookList` component should display books list in html table using **the <code>Book</code> component** <a href="https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/BooksList/BooksAsTable.png">screenshot from pathwright</a>
- [x] You should then pass a book object as a prop to each Book component

File: src/actions/index.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  }
+);
+
+const removeBook = (id) => (
+  {
+    type: 'REMOVE_BOOK',
+    id,
+  }
+);
+
```
Comment (teekaytech) [inline src/actions/index.js]: - [x] The `src/actions/index.js` script should contain <code>CREATE_BOOK</code> and <code>REMOVE_BOOK</code> actions **with Book objects as an argument** <a href="https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/ReduxActions/ReducxActions.png">screenshot from pathwright</a>

- Change the `removeBook` function argument to book object, and not id.

File: package.json [hunk 4]
```diff
@@ -15,10 +15,10 @@
...
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "start": "DISABLE_ESLINT_PLUGIN=true react-scripts start",
+    "build": "DISABLE_ESLINT_PLUGIN=true react-scripts build",
+    "test": "DISABLE_ESLINT_PLUGIN=true react-scripts test",
+    "eject": "DISABLE_ESLINT_PLUGIN=true react-scripts eject"
```
Comment (teekaytech) [inline package.json]: - [x] Disabling ESlint rules at this level is not advised. Fix the linter issues as much as you can, use inline or page disabling of ESLint rules where you have to.

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__I must commend your efforts so far in this project. Here are the major highlights for me:__

- Pull Request has a meaningful title and a brief description :+1: 
- Commit messages are also great :+1: 

__However, some changes are required before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] The `src/reducers/index.js` should combine app-specific reducers.
- [x] Please check my inline comments below for other required changes.

**[ OPTIONAL CHANGES ]**

- I will suggest you separate the sample books object (initialState) from the books reducer. You can move the object to a separate file and import it anywhere you need it.

* * *
**Please make the required changes and submit another review request when you are done.**

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review unless it is requested otherwise.**

**Always post your inquiries and questions on this PR thread as a comment. You can mention me (@teekaytech) so I can see your comments and respond.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by RNtaate** [comment]
## Changes made
- [X] Combined app-specific reducers in `src/reducers/index.js`
- [X] Moved initialState from books reducer to a separate file.
- [X] Updated Book.js component to take a `Book` object as prop and represent a single table row of books
- [X] Updated BooksList component to display the books list in HTML table using the **Book** component
- [X] Updated `removeBook` action function to take a book object as argument.
- [X] Re-enabled disabled eslint rules.

**Review by BrittanyBlake — APPROVED** [review]
### Status: Approved ✅ 
****
### Hi Team! 👋 
### Great job implementing the requested changes!! 🎖️ It's a 👍 from me! Go ahead and merge this pull request and keep up the good work 😄
****


 Good Luck and Happy Coding! 🍀 👩🏾‍💻

**Pull request merged**